backend/src/prisma/recipes.js

```diff
@@ -1,29 +1,133 @@
-const { PrismaClient } = require('@prisma/client')
+const Prisma = require("prisma");
+const { DeleteRecipeError } = require('../errors');
+const { PrismaClient } = require('@prisma/client');
 
 const prisma = new PrismaClient()
 
 
 async function getRecipes() {
-    return await prisma.recipe.findMany({
-        include: {
-            RecipeIngredient: true,
-        },
-    });
+    let recipes = await prisma.recipe.findMany();
+    // console.log("recipes:");
+    // console.log(recipes);
+    for (let i = 0; i < recipes.length; i++) {
+        let recipeIngredients = await prisma.recipeIngredient.findMany({
+            where: {
+                recipeId: recipes[i].id,
+            }
+        });
+        // console.log("recipeIngredients:");
+        // console.log(recipeIngredients);
+        let ingredientList = [];
+        let ing;
+        for (let j = 0; j < recipeIngredients.length; j++) {
+            ing = await prisma.ingredient.findUnique({
+                where: {
+                    id: recipeIngredients[j].ingredientId,
+                }
+            });
+            // console.log("ingredients:");
+            // console.log(ing);
+            ing.amount = recipeIngredients[j].amount;
+            ingredientList.push(ing);
+        }
+        // console.log("ingredientList:");
+        // console.log(ingredientList);
+        recipes[i].ingredients = ingredientList;
+    }
+    console.log("recipes:");
+    console.log(recipes);
+    return recipes;
 }
 
-
-async function createRecipe() {
-    return null
+async function createRecipe(recipe) {
+    let createdRecipe = await prisma.recipe.create({
+        data: {
+            name: recipe.name,
+            description: recipe.description,
+            preparation_length: recipe.preparationLength,
+            final_amount: recipe.ingredients.length
+        },
+    });
+    for (let i = 0; i < recipe.ingredients.length; i++) {
+        await prisma.recipeIngredient.create({
+            data: {
+                recipeId: createdRecipe.id,
+                ingredientId: recipe.ingredients[i].id,
+                amount: recipe.ingredients[i].amount
+            },
+        });
+    }
+    return createdRecipe;
 }
 
 
-async function updateRecipe() {
-    return null
+async function updateRecipe(id, req) {
+    try {
```

Comment:
Try / catch block should contain the smallest amount of code and catch a specific error.